rviz_default_plugins/src/rviz_default_plugins/displays/screw/screw_display.cpp

```diff
@@ -0,0 +1,243 @@
+
+  history_length_property_->setMin(1);
+  history_length_property_->setMax(100000);
+}
+
+template<class MessageType>
+void ScrewDisplay<MessageType>::onInitialize()
+{
+  rviz_common::MessageFilterDisplay<MessageType>::onInitialize();
+  updateHistoryLength();
+}
+
+// Override rviz::Display's reset() function to add a call to clear().
+template<class MessageType>
+void ScrewDisplay<MessageType>::reset()
+{
+  rviz_common::MessageFilterDisplay<MessageType>::reset();
+  visuals_.clear();
+}
+
+template<class MessageType>
+void ScrewDisplay<MessageType>::updateProperties()
+{
+  float alpha = alpha_property_->getFloat();
+  float linear_scale = linear_scale_property_->getFloat();
+  float angular_scale = angular_scale_property_->getFloat();
+  float width = width_property_->getFloat();
+  bool hide_small_values = hide_small_values_property_->getBool();
+  Ogre::ColourValue linear_color = linear_color_property_->getOgreColor();
+  Ogre::ColourValue angular_color = angular_color_property_->getOgreColor();
+
+  for (size_t i = 0; i < visuals_.size(); i++) {
+    visuals_[i]->setLinearColor(linear_color.r, linear_color.g, linear_color.b, alpha);
+    visuals_[i]->setAngularColor(angular_color.r, angular_color.g, angular_color.b, alpha);
+    visuals_[i]->setLinearScale(linear_scale);
+    visuals_[i]->setAngularScale(angular_scale);
+    visuals_[i]->setWidth(width);
+    visuals_[i]->setHideSmallValues(hide_small_values);
+  }
+}
+
+// Set the number of past visuals to show.
+template<class MessageType>
+void ScrewDisplay<MessageType>::updateHistoryLength()
+{
+  while (visuals_.size() > static_cast<size_t>(history_length_property_->getInt())) {
+    visuals_.pop_front();
+  }
+}
+
+// This is our callback to handle an incoming message.
+template<class MessageType>
+void ScrewDisplay<MessageType>::processMessagePrivate(
+  const std_msgs::msg::Header & header,
+  const geometry_msgs::msg::Vector3 & linear,
+  const geometry_msgs::msg::Vector3 & angular)
+{
+  if (!(rviz_common::validateFloats(linear) && rviz_common::validateFloats(angular))) {
+    rviz_common::MessageFilterDisplay<MessageType>::setStatus(
+      rviz_common::properties::StatusProperty::Error, "Topic",
+      "Message contained invalid floating point values (nans or infs)");
+    return;
+  }
+
+  // Here we call the rviz::FrameManager to get the transform from the
+  // fixed frame to the frame in the header of this Imu message.  If
+  // it fails, we can't do anything else so we return.
+  Ogre::Quaternion orientation;
+  Ogre::Vector3 position;
+  if (!rviz_common::MessageFilterDisplay<MessageType>::context_->getFrameManager()->getTransform(
+      header.frame_id, header.stamp, position,
+      orientation))
+  {
+    RVIZ_COMMON_LOG_DEBUG_STREAM(
+      "Error transforming from frame '" << header.frame_id.c_str() << "' to frame '" <<
+        qPrintable(rviz_common::MessageFilterDisplay<MessageType>::fixed_frame_) << "'");
```

Comment:
I'm not entirely sure if this is better, but it might be worth considering setting the status as opposed to printing something to the console here.